### PR TITLE
Fix ModalSubmitInteraction Typings

### PR DESCRIPTION
**Pull request by sasial-dev**
**Please describe the changes this PR makes and why it should be merged:**

Fix typings not existing for already existing code.

https://github.com/discordjs/discord.js/blob/v13/src/structures/ModalSubmitInteraction.js#L47-L51

**Status and versioning classification:**

- Code changes have been tested against the Discord API, or there are no code changes
- I know how to update typings and have done so, or typings don't need updating

### CODE REVIEW

**Comment by Jiralite** [comment]
> I believe it should not have a `message` property at all
> 
> https://discord.com/developers/docs/interactions/receiving-and-responding#interaction-object-interaction-structure
> 
> > for components



<details>
<summary>Modals do receive the payload if it was from a component!</summary>

```JavaScript
{
  t: 'INTERACTION_CREATE',
  s: 4,
  op: 0,
  d: {
    version: 1,
    type: 5,
    token: 'token',
    message: {
      type: 0,
      tts: false,
      timestamp: '2022-02-16T19:09:35.384000+00:00',
      pinned: false,
      mentions: [],
      mention_roles: [],
      mention_everyone: false,
      id: '943584917885960303',
      flags: 0,
      embeds: [],
      edited_timestamp: null,
      content: 'Do you have any feedback for us? Do you wish to alert us of misbehaviour within the friends chat or Discord server? <#942814460769366057> is the place to be!\n' +
        '\n' +
        "If you are attempting to report an ongoing situation that needs immediate attention, feel free to mention the <@&886323824822456357> role and we'll respond accordingly.\n" +
        '\n' +
        'Please interact with the respective button to get started!',
      components: [Array],
      channel_id: '942814460769366057',
      author: [Object],
      attachments: []
    },
    member: {
      user: [Object],
      roles: [Array],
      premium_since: null,
      permissions: '4398046511103',
      pending: false,
      nick: null,
      mute: false,
      joined_at: '2021-09-11T18:54:26.945000+00:00',
      is_pending: false,
      flags: 0,
      deaf: false,
      communication_disabled_until: null,
      avatar: null
    },
    locale: 'en-GB',
    id: '963055469386149944',
    guild_locale: 'en-US',
    guild_id: '886323824801488996',
    data: { custom_id: 'MAIL_SOULOBBY_REPORT', components: [Array] },
    channel_id: '942814460769366057',
    application_id: '559093467264057344'
  }
}
```
</details>

This pull request is a backport of #7705 anyhow.

File: typings/index.d.ts [hunk 1]
```diff
@@ -1913,6 +1913,7 @@ export class ModalSubmitInteraction<Cached extends CacheType = CacheType> extend
   public components: PartialModalActionRow[];
   public deferred: boolean;
   public ephemeral: boolean | null;
+  public message: GuildCacheMessage<Cached>;
```
Comment (Mateo-tem) [inline typings/index.d.ts:1916]:
```suggestion
  public message: GuildCacheMessage<Cached> | null;
```

**Review by Mateo-tem — CHANGES_REQUESTED** [review]

**Comment by Jiralite** [comment]
Hi @sasial-dev, can you look at #7705 and made the necessary changes here? It should match! Seems there is only one change to do.

**Comment by sasial-dev** [comment]
According to my local copy, the 2nd change in #7705 already exists!

**Comment by sasial-dev** [comment]
Already exists.
https://github.com/sasial-dev/discord.js/blob/1563bb54eeb7475017c7113450cf7f112323deb7/typings/index.d.ts#L1900

**Comment by Jiralite** [comment]
I said you need to modify it! Look at the upstream pull request and compare.

**Comment by sasial-dev** [comment]
Apologies for the misunderstanding - all sorted :)

**Comment by sasial-dev** [comment]
Stuffed up the rebase, sorry. I'll get rid of them

**Review by Jiralite — APPROVED** [review]

**Comment by sasial-dev** [comment]
^ Rebased through Github UI, seemed to have fixed it (but didn't sign)

**Comment by sasial-dev** [comment]
@Mateo-tem, can you re-review?

**Review by vladfrangu — APPROVED** [review]

**Review by SpaceEEC — APPROVED** [review]

**Review by Mateo-tem — APPROVED** [review]

**Review by iCrawl — APPROVED** [review]

**Pull request merged**